### DISCUSSION TITLE
Headless CMS | Community Edition

### DISCUSSION THREAD

**Discussion by solverat** [pimcore/pimcore#14230] [General]
Hello @ all.

We are currently in an evaluation phase where we are testing 100% headless availability in PIMCORE.

The biggest problem we have is the CMS function. Not the frontend area (DataHub for the win), but the FE-Editmode.

Two Things:

## ExtJs
The editmode is managed by ExtJs. Apart from this complicated circumstance, there is still this elephant in the room, which is heavily [discussed](https://github.com/pimcore/pimcore/discussions/12795), and also [technically](https://github.com/pimcore/pimcore/issues/12094) already precautions are made: Still, no one knows what exactly is happening.

## Pimcore Headless Documents Bundle
This bundle generally solves all the problems, if it weren't for one small thing: It only exists as an [enterprise bundle](https://pimcore.com/docs/headless-documents/current/index.html). Although the idea started in the [open source world](https://github.com/pimcore/pimcore/issues/8197), it is now only available through the Enterprise license.

I'm starting this discussion because one way or another, as a vendor, we will or must build a solution for our customers. We haven't made a decision about the license yet, however, I think it would be desirable if we could discuss a few things first.

### Is it cheeky to say that the PHD Bundle actually belongs in the core? 
- Headless content is mostly standard in 2023: There are more and more headless CMS frameworks (keyword "Strapi") coming up, that will eventually outpace the CMS part of PIMCORE - Our opinion: this feature belongs in a current standard - just like car tires belonged to the standard when buying a car
- The bundle could be at least a single paid bundle, because the Enterprise license actually manages a completely different universe than managing headless CMS pages
- A further development by the community also reflects positively on PIMCORE

### ExtJs?
- This discussion is going on (see link above), there is no answer so far
- If there would be at least a little hint (keyword "technology"), we and probably many others could better plan their future with PIMCORE

Finally, I'd like to say that it would be almost sad and most certainly an incredible waste of time if we end up developing a second bundle that is built more or less the same as an existing one.

@brusch: We would love to hear your opinion on this. 

Thanks!

**Comment by dpfaffenbauer** [comment]
> Headless content is mostly standard in 2023: There are more and more headless CMS frameworks (keyword "Strapi") coming up, that will eventually outpace the CMS part of PIMCORE - Our opinion: this feature belongs in a current standard - just like car tires belonged to the standard when buying a car

I don't agree on that. Headless is a trend, just like SPAs were a few years ago (JS world is moving super fast and so are the trends). For most of the Projects, it doesn't make a difference if Headless or Request/Response with HTML. Having no "free and Open Source" Pimcore Headless CMS, will not kill Pimcore immediately.

<sup>satire on </sup>
(car tires in the future will probably end up needing a subscription model to get :P)
<sub> satire off </sub>

> A further development by the community also reflects positively on PIMCORE

That is true, but with the PEM Agreement, you can give the development, back to the Enterprise Community. (at least)

> ExtJs?

ExtJs will mostly be a thing of the past with Pimcore 11/12. I don't know what it's gonna be, VueJs, ReactJs, Angular or whatever. But the decision clearly has been made. It's more a question of when rather than if.

But: ExtJS still is a great framework. It made Pimcore what it is today, and if it wasn't for ExtJs, Pimcore would have never turned out the way it is. Only Problem with ExtJs: Lack of Developers knowing it, f***ing licensing. 

But as time progresses, it has to go. I am also waiting for what is going to happen.

> Finally, I'd like to say that it would be almost sad and most certainly an incredible waste of time if we end up developing a second bundle that is built more or less the same as an existing one.

Would it though? I did the same with eCommerce for Pimcore and it turned out better than what Pimcore has. Problem rather is: As a Pimcore Partner, you are not allowed to create new Bundles that compete with Enterprise Bundles. (At least you are not allowed to open source it, AFAIK)

I don't think opinions really matter in that case. It has been a clear Business decision to make it Enterprise. At the end of the day, Pimcore's staff also need to buy supper.

**Comment by solverat** [comment]
Headless is not a Trend. It is also not necessarily connected with JS. It's about structured separation of technologies and skills. And it won't go away. 

I'm not talking about if ExtJs is good or bad. Personally, I could live with it the next 20 years. I brought it up because it has been announced by PIMCORE itself. And besides presumptions from your side, there is no other real statement. 

At the end, it is all about the future of the CMS Part of PIMCORE. If they see future-proof development only in the enterprise world, that's okay for me. That's why I'm writing this post.

However, as I mentioned it in my previous post, currently, no decision from our side regarding license has been made. If it is going to be closed source, fine (the main thing is that we have a solution for our customers without EL).

**Comment by dpfaffenbauer** [comment]
> Headless is not a Trend. It is also not necessarily connected with JS. It's about structured separation of technologies and skills. And it won't go away.

I never said it would go away. Headless is here to stay, same as SPA and other technologies. I was saying it is not necessary a must have feature for the future of Pimcore. A Corporate Website, ecommerce,  portals or whatever also work perfectly fine with classical Request/Response.

> I'm not talking about if ExtJs is good or bad. Personally, I could live with it the next 20 years. I brought it up because it has been announced by PIMCORE itself. And besides presumptions from your side, there is no other real statement.

It's not just "presumptions". I talked a lot about that to Pimcore. Back then, they didn't even know, but tended to these frameworks. Will see what it will be. Maybe Pimcore already knows, maybe not. Doesn't matter until we see something.

> At the end, it is all about the future of the CMS Part of PIMCORE. If they see future-proof development only in the enterprise world, that's okay for me. That's why I'm writing this post.

Pimcore CMS is not dead, and the future is not just about Headless. 

BTW: there is a solution you can already use: DataHub supports Documents. So you can get them headless with that. The Headless Bundle just renders some simple Twig HTML for editing in Pimcore. It uses what Pimcore CMS already offers and doesn't extend it by much. (Only some configuration that is parsed and used by document types/bricks or something like that)
